plugins/statistics.koplugin/main.lua

```diff
@@ -2636,4 +2748,124 @@ function ReaderStatistics:getCurrentBookReadPages()
     return read_pages
 end
 
+function ReaderStatistics.onSync(local_path, cached_path, income_path)
+    local conn_income = SQ3.open(income_path)
+    local ok1, v1 = pcall(conn_income.rowexec, conn_income, "PRAGMA schema_version")
+    if not ok1 or tonumber(v1) == 0 then
+        -- no income db or wrong db, first time sync
+        logger.warn("statistics open income DB failed", v1)
+        return true
+    end
+
+    local sql = "attach '" .. income_path:gsub("'", "\\'") .."' as income_db;"
```

Comment:
(Not familiar with our Cloud storage code: where will this income_db be located in the koreader/ subdirectories ?)

Reply:
It will be in the same folder as the local database, then be deleted after the sync is done.